iceoryx_dust/include/iceoryx_dust/cxx/std_chrono_support.hpp

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2023 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+#ifndef IOX_DUST_STD_CHRONO_SUPPORT_HPP
+#define IOX_DUST_STD_CHRONO_SUPPORT_HPP
+
+#include "iox/duration.hpp"
+#include "iox/into.hpp"
+
+#include <chrono>
+
+namespace iox
+{
+/// @brief Construct a Duration object from std::chrono::nanoseconds
+/// @param[in] value as nanoseconds
+/// @attention Since negative durations are not allowed, the duration will be clamped to 0
+template <>
+struct FromImpl<std::chrono::nanoseconds, units::Duration>
+{
+    static units::Duration fromImpl(const std::chrono::nanoseconds& value) noexcept;
+};
```

Comment:
Nothing for now but something for an interested contributor. I think all four specializations could be combined into one with a partial specialization for `std::chrono::duration<Rep, Period>`. Although, this might also open a can of worms.